guides/user/managed-delivery/environment-constraints/index.md

```diff
@@ -0,0 +1,316 @@
+analysis, or launching a smoke test pipeline against a prior environment, gating promotion on its success.
+The implicit ordering by type ensures that a pipeline constraint on a production environment intended to
+run tests against a prior environment will not be launched until the artifact version being evaluated has
+actually been delivered into that prior environment. All stateful constraints can be manually overridden,
+such as when a canary failure is deemed to be expected for a given change, or a false positive.
+
+When an environment defines multiple constraints, all must pass before a new artifact will be promoted into
+the environment.
+ 
+## Available Constraints
+
+### Depends On
+
+The `depends-on` constraint ensures that an artifact version will not deploy into the environment it is
+attached to, unless it has successfully deployed into the dependent environment.
+
+**Parameters**
+`environment`: The name of the dependent environment. It must be defined within the same delivery config as
+the target environment containing the constraint.
+
+**Example**
+
+```yaml
+  constraints: 
+  - type: depends-on
+    environment: testing
+```
+
+### Allowed Times
+
+The `allowed-times` constraints provides a time based gate on promotion into an environment. It can be used to
+ensure promotion to an environment only occurs during business hours, or quiet times.
+
+**Parameters**
+`windows`: A list of time windows; multiple windows can be specified on a single constraint. The constraint is
+satisfied if the current time matches any single window.
+`tz`: *Optional* The timezone to evaluate the constraint in. This is optional, by default the timezone specified via
+the `default.time-zone` spring configuration property is used, not the JVM or OS timezone. If your organization
+operates in a single timezone, this property should likely be set that timezone.
+
+**Window Parameters**
+`hours`: Supports ranges (i.e. `10-20`) and comma separated lists (i.e. `5,6,7,8`) or a combination
+of the two.
+`days`: Supports aliases (`weekdays` and `weekends`), as well as ranges and comma separated lists of
+days in their java standard (`java.util.Calendar`) short (`mon-thu`) or full (`monday-thursday`) names.
+
+**Example**
+```yaml
+  constraints:
+  - type: allowed-times
+    windows:
+      - days: Monday-Wednesday,Friday
+        hours: 8-16
+      - days: Thursday
+        hours: 14-16
+```
+This allows artifacts to deploy into the environment between 8am-4pm on Monday, Tuesday, Wednesday, and Friday or
+between 2-4pm on Thursday.
+
+### Manual Judgement
+
+The `manual-judgement` constraint prevents an artifact from deploying to an environment without explicit
+approval via an external API client. UI support for approving or rejecting judgements as well as an 
+interactive slack bot are under development but direct use of Spinnaker's REST API is currently required to
+answer a manual judgement.
```

Comment:
Worth providing that endpoint and a sample request? Or not, because we will be adding actual support for it soon?

Reply:
Ah I see that it's below. It would be helpful to link to that section right here!

Reply:
Good call, added the link.